src/main/java/org/apache/mesos/hdfs/state/PersistentState.java

```diff
@@ -71,6 +72,22 @@ public void setFrameworkId(FrameworkID frameworkId) throws InterruptedException,
     zkState.store(value).get();
   }
 
+  public void resetDeadNodeTimeStamps() {
+    Date date = DateUtils.addSeconds(new Date(), conf.getDeadNodeTimeout());
+
+    if (getDeadJournalNodes().size() > 0) {
+      deadJournalNodeTimeStamp = new Timestamp(date.getTime());
```

Comment:
TODO: Create a timestamp per deadJN/NN? As long as one JN keeps dying within the timeout, we won't have a chance to expire the other dead nodes that have been dead for longer than the timeout. Once we remove the deadDN logic, that only leaves up to 5 timeouts to track.